catalog/spatial/registry/registry-schema-bindings/src/test/java/org/codice/ddf/registry/schemabindings/RegistryPackageWebConverterTest.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.registry.schemabindings;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.bind.JAXBElement;
+
+import org.codice.ddf.parser.Parser;
+import org.codice.ddf.parser.ParserConfigurator;
+import org.codice.ddf.parser.ParserException;
+import org.codice.ddf.parser.xml.XmlParser;
+import org.junit.Before;
+import org.junit.Test;
+
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryObjectType;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+
+public class RegistryPackageWebConverterTest {
+    private Parser parser;
+
+    private ParserConfigurator configurator;
+
+    @Before
+    public void setUp() {
+        parser = new XmlParser();
+
+        configurator = parser.configureParser(Arrays.asList(RegistryObjectType.class.getPackage()
+                        .getName(),
+                RegistryPackageUtils.OGC_FACTORY.getClass()
+                        .getPackage()
+                        .getName(),
+                RegistryPackageUtils.GML_FACTORY.getClass()
+                        .getPackage()
+                        .getName()),
+                this.getClass()
+                        .getClassLoader());
+    }
+
+    @Test
+    public void testRoundTrip() throws Exception {
+        RegistryObjectType registryObject = getRegistryObjectFromResource(
+                "/csw-full-registry-package.xml");
+
+        Map<String, Object> registryMap = RegistryPackageWebConverter.getRegistryObjectWebMap(
+                registryObject);
+
+        RegistryObjectType convertedRegistryObject =
+                RegistryPackageWebConverter.getRegistryPackageFromWebMap(registryMap);
+
+        assertThat(registryObject.getObjectType(),
```

Comment:
Did you try to do `assertThat(convertedRegistryObject, equalTo(registryObject))`? All those generated classes have `equals()` methods and should be able to do deep compares.

If that doesn't work, it may be worth trying to marshall both registry objects and compare their outputs to make sure that all the values are properly converted.